src/message_type.rs

```diff
@@ -17,7 +17,152 @@
+            message: PbftMessageWrapper::PbftMessage(message),
+            message_bytes: vec![],
+            header_bytes: vec![],
+            header_signature: vec![],
+        }
+    }
+
+    pub fn from_view_change_message(message: PbftViewChange) -> Self {
+        Self {
+            header_bytes: vec![],
+            header_signature: vec![],
+            message: PbftMessageWrapper::PbftViewChange(message),
+            message_bytes: vec![],
+        }
+    }
+
+    pub fn get_info(&self) -> &PbftMessageInfo {
+        match &self.message {
+            PbftMessageWrapper::PbftMessage(m) => &m.get_info(),
+            PbftMessageWrapper::PbftViewChange(m) => &m.get_info(),
+        }
+    }
+
+    pub fn set_info(&mut self, info: PbftMessageInfo) {
+        match self.message {
+            PbftMessageWrapper::PbftMessage(ref mut m) => m.set_info(info),
+            PbftMessageWrapper::PbftViewChange(ref mut m) => m.set_info(info),
+        }
+    }
+
+    pub fn get_block(&self) -> &PbftBlock {
+        match &self.message {
+            PbftMessageWrapper::PbftMessage(m) => m.get_block(),
+            PbftMessageWrapper::PbftViewChange(_) => {
+                panic!("ParsedPeerMessage.get_block found a view change message!")
+            }
+        }
+    }
+
+    pub fn get_pbft_message(&self) -> &PbftMessage {
+        match &self.message {
+            PbftMessageWrapper::PbftMessage(m) => m,
+            PbftMessageWrapper::PbftViewChange(_) => {
+                panic!("ParsedPeerMessage.get_pbft_message found a view change message!")
+            }
+        }
+    }
+
+    pub fn get_view_change_message(&self) -> &PbftViewChange {
+        match &self.message {
+            PbftMessageWrapper::PbftMessage(_) => {
+                panic!("ParsedPeerMessage.get_view_change_message found a pbft message!")
+            }
+            PbftMessageWrapper::PbftViewChange(m) => m,
+        }
+    }
+
+    pub fn from_peer_message(message: PeerMessage) -> Result<Self, PbftError> {
+        verify_peer_message_hash(&message.content, &message.header.content_sha512)?;
+
+        let parsed_message = match protobuf::parse_from_bytes::<PbftMessage>(&message.content) {
+            Ok(m) => PbftMessageWrapper::PbftMessage(m),
+            Err(_) => match protobuf::parse_from_bytes::<PbftViewChange>(&message.content) {
+                Ok(m) => PbftMessageWrapper::PbftViewChange(m),
+                Err(_) => return Err(PbftError::InternalError("Couldn't parse message!".into())),
+            },
+        };
+
+        Ok(Self {
+            header_bytes: message.header_bytes,
+            header_signature: message.header_signature,
+            message: parsed_message,
+            message_bytes: message.content.clone(),
+        })
+    }
+
+    pub fn from_bytes(message: Vec<u8>) -> Result<Self, PbftError> {
+        let peer_message = PeerMessage {
+            content: message,
+            header_bytes: b"myself".to_vec(),
```

Comment:
@aludvik: This is here due to the fact that the `ConsensusPeerMessage` type should be treated as opaque and not constructed by PBFT code, but we still need to calculate the primary's approval of the consensus seal. Would it make sense to instead just assume that the primary is OK with the block, and not include its vote in the consensus seal, and require `2f` votes?